pdns/sdig.cc

```diff
@@ -371,19 +371,29 @@ try {
         }
         auto traceIDArg = std::string(argv[++i]);
         pdns::trace::TraceID traceid;
+        pdns::trace::SpanID spanid;
         if (traceIDArg == "-") {
           traceid.makeRandom();
+          spanid.makeRandom();
```

Comment:
Don't we want the default case to be: random trace id, empty (absent) span id? As the spanid ends up being the parent of e.g. the main recursor span, so with no parent span id set, the main span of rec becomes the root span.

Reply:
My point being: sdig itself does not create spans and it is an initiator, so main span the next server creates should be the root span, i.e. have no patent span id.